### ISSUE TITLE
Terraform provider for uptime monitoring

### ISSUE THREAD

**Issue by mj-sakellaropoulos** [OneUptime/oneuptime#514]
**Is your feature request related to a problem? Please describe.**
I am currently evaluating Uptime/Status page products for my employer (Replacing Freshping). We have atleast 40+ monitors and individual status pages to implement. This is not really feasible via UI. Ideally, we want to do this with an IaC approach.

**Describe the solution you'd like**
Having an Official Terraform provider for your API would be ideal and help us to adopt your software.

**Describe alternatives you've considered**
- Other products with terraform providers

**Additional context**
At the very least, an OpenAPI definition file would help

I am mentioning uptime monitoring here as it is my current priority, but we also have use cases for Terraform for the Incidents/Oncall APIs.

**Comment by simlarsen** [comment]
Yep. On our roadmap. If you are or willing to be  an enterprise customer with a contract - we're more than happy to move this to the top of the list for you.

**Comment by benjimin** [comment]
It's been a year, any update or ETA? (Or how much for the enterprise contract to get it across the line?)

We currently use UptimeRobot for branded status pages, simple website (polled request) monitoring, and starting to use heartbeat (aka. cron or webhook) monitors. We want to terraform this because:
* IaC makes it more feasible to maintain dozens of monitors consistently.
* CI/CD avoids needing to manage admin team access to the platform (onboarding/offboarding staff, SSO setup, credentials, granular permissions, change review/approval procedures, etc).
* Terraform/OpenTofu naturally automates passing all the different webhook url strings to all the corresponding cron-like workloads (such as terraformed AWS Lambdas and Kubernetes CronJobs), instead of us needing to manually paste magic strings all over the place.

Unfortunately, there is no official terraform provider for UptimeRobot, and several unofficial efforts all seem to have stalled before releasing heartbeat support. It looks like maybe the closest equivalent would be to instead migrate to BetterStack, but if migrating anyway then it would be preferable to lend support to an Open Source equivalent.

It seems like it ought be fairly trivial to write a Terraform provider, given that there is a documented API?

**Comment by simlarsen** [comment]
@benjimin  If you're willing to be an enterprise customer, we can build this for you.
